src/backend/InvenTree/plugin/builtin/exporter/bom_exporter.py

```diff
@@ -0,0 +1,290 @@
+"""Multi-level BOM exporter plugin."""
+
+from django.utils.translation import gettext_lazy as _
+
+import rest_framework.serializers as serializers
+
+from part.models import BomItem
+from plugin import InvenTreePlugin
+from plugin.mixins import DataExportMixin, SettingsMixin
+
+
+class BomExporterOptionsSerializer(serializers.Serializer):
+    """Custom export options for the BOM exporter plugin."""
+
+    export_levels = serializers.IntegerField(
+        default=1,
+        label=_('Levels'),
+        help_text=_('Number of levels to export'),
+        min_value=0,
+    )
+
+    export_stock_data = serializers.BooleanField(
+        default=True, label=_('Stock Data'), help_text=_('Include part stock data')
+    )
+
+    export_pricing_data = serializers.BooleanField(
+        default=True, label=_('Pricing Data'), help_text=_('Include part pricing data')
+    )
+
+    export_supplier_data = serializers.BooleanField(
+        default=True, label=_('Supplier Data'), help_text=_('Include supplier data')
+    )
+
+    export_manufacturer_data = serializers.BooleanField(
+        default=True,
+        label=_('Manufacturer Data'),
+        help_text=_('Include manufacturer data'),
+    )
+
+    export_substitute_data = serializers.BooleanField(
+        default=True,
+        label=_('Substitute Data'),
+        help_text=_('Include substitute part data'),
+    )
+
+    export_parameter_data = serializers.BooleanField(
+        default=True,
+        label=_('Parameter Data'),
+        help_text=_('Include part parameter data'),
+    )
+
+
+class BomExporterPlugin(DataExportMixin, SettingsMixin, InvenTreePlugin):
+    """Builtin plugin for performing multi-level BOM exports."""
+
+    NAME = 'BOM Exporter'
+    SLUG = 'bom-exporter'
+    TITLE = _('Multi-Level BOM Exporter')
+    DESCRIPTION = _('Provides support for exporting multi-level BOMs')
+    VERSION = '1.0.0'
+    AUTHOR = _('InvenTree contributors')
+
+    SETTINGS = {}
```

Comment:
Why are settings included when they are not used?

Reply:
Was planning on having some settings for this plugin but no longer needed. I will revert this.